mds/apis/agency_api/v0_x/vehicles.py

```diff
@@ -92,9 +94,15 @@ class DeviceRegisterSerializer(serializers.Serializer):
     def create(self, validated_data):
         provider_id = self.context["request"].user.provider_id
         try:
-            return models.Device.objects.create(
+            device = models.Device.objects.create(
                 provider_id=provider_id, **validated_data
             )
+            models.EventRecord.objects.create(
+                timestamp=datetime.datetime.now(datetime.timezone.utc),
+                device_id=device.id,
```

Comment:
Now you have the device object, you could write `device=device`, but this is nitpicking.